src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -215,6 +212,45 @@ private static bool ForceWriteTargets()
             return buildGroupSets;
         }
 
+        private static IReadOnlyList<GraphItem<RemoteResolveResult>> TopologicalSort(IEnumerable<GraphItem<RemoteResolveResult>> items)
+        {
+            var sorted = new List<GraphItem<RemoteResolveResult>>();
+            var toSort = items.Distinct().ToList();
+
+            while (toSort.Count > 0)
+            {
+                // Order packages by parent count, take the child with the lowest number of parents
+                // and remove it from the list
+                var nextPackage = toSort.OrderBy(item => GetParentCount(toSort, item.Key.Name))
+                    .ThenBy(item => item.Key.Name, StringComparer.OrdinalIgnoreCase).First();
+
+                sorted.Add(nextPackage);
+                toSort.Remove(nextPackage);
+            }
+
+            // the list is ordered by parents first, reverse to run children first
+            sorted.Reverse();
```

Comment:
Use toSort.OrderByDescending instead of reversing here. It's not really needed, and it also means that alphabetical order will be reversed.